analyzers/VMRay/vmray.py

```diff
@@ -7,98 +7,230 @@
+                    value += " using tactics: {}".format(
+                        ", ".join(mitre_technique["tactics"])
+                    )
+                if predicate:
+                    taxonomies.append(
+                        self.build_taxonomy(level, self._namespace, predicate, value)
+                    )
+
+            # add child sample taxonomies if they have been added
+            taxonomies.extend(
+                self._taxonomies_for_samples(sample.get("sample_child_samples", []))
+            )
+        return taxonomies
 
     def summary(self, raw):
         taxonomies = []
-        namespace = "VMRay"
-        predicate = "Scan"
-
-        r = {
-            'reports': []
-        }
-
-        if raw.get('scanreport', None) and len(raw.get('scanreport').get('data')) > 0:
-            for scan in raw.get('scanreport').get('data'):
-                r['reports'].append({
-                    'score': scan.get('sample_score'),
-                    'sample_severity': scan.get('sample_severity'),
-                    'sample_last_reputation_severity': scan.get('sample_last_reputation_severity'),
-                    'url': scan.get('sample_webif_url')
-                })
-
-        if len(r["reports"]) == 0:
-            value = "No Scan"
-            level = "info"
-            taxonomies.append(self.build_taxonomy(level, namespace, predicate, value))
+        samples = raw.get("samples", [])
+        if len(samples) == 0:
+            taxonomies.append(
+                self.build_taxonomy("info", self._namespace, "None", "No Scan")
+            )
         else:
-            for s in r["reports"]:
-                i = 1
-                if s["sample_severity"] == "not_suspicious":
-                    level = "safe"
-                elif s["sample_severity"] == "malicious":
-                    level = "malicious"
-                else:
-                    level = "info"
-
-                if len(r["reports"]) > 1:
-                    value = "{}( from scan {})".format(s["score"], i)
-                else:
-                    value = "{}".format(s["score"])
-                taxonomies.append(self.build_taxonomy(level, namespace, predicate, value))
-                i += 1
-
+            taxonomies.extend(self._taxonomies_for_samples(samples))
         return {"taxonomies": taxonomies}
 
+    def _artifacts_for_samples(self, samples):
+        artifacts = []
+        for sample in samples:
+            link = sample.get("sample_webif_url", None)
+            iocs = sample.get("sample_iocs", {}).get("iocs", {})
+
+            for ioc_type, ioc_payload_name in self._ioc_mapping.items():
+                if ioc_type in iocs:
+                    for ioc_node in iocs[ioc_type]:
+                        severity = ioc_node.get("severity", "")
+                        level = self._severity_mapping.get(severity, "info")
+                        tags = [severity, level, ioc_node["type"]]
```

Comment:
My original PR used a `set` on purpose to avoid duplicate tags since `severity` can times be equal to `level`.